packages/transport/src/core/Reference.ts

```diff
@@ -6,43 +6,67 @@ import { TransportCoreErrors } from "./TransportCoreErrors";
 
 export interface IReference extends ISerializable {
     id: ICoreId;
+    backboneBaseUrl?: string;
     key: ICryptoSecretKey;
+    forIdentityTruncated?: string;
+    passwordType?: number;
 }
 
 export class Reference extends Serializable implements IReference {
-    @validate()
+    @validate({ regExp: new RegExp("^[A-Za-z0-9]{20}$") })
     @serialize()
     public id: CoreId;
 
+    @validate({ nullable: true })
+    @serialize()
+    public backboneBaseUrl?: string;
+
     @validate()
     @serialize()
     public key: CryptoSecretKey;
 
+    @validate({ nullable: true, regExp: new RegExp("^[0-9a-f]{4}$") })
+    @serialize()
+    public forIdentityTruncated?: string;
```

Comment:
why not 
```
    public forIdentity?: string;
```

I don't really understand where the word Truncated comes from here.

Reply:
because we only give the last digits. Wasn't sure if there would be misunderstandings, am fine with both

Reply:
> last digits

chars and digits, right?

Reply:
it's a hexadecimal - so you can spin it both ways

Reply:
True. I'm ok with truncated in this case, now I understand it.